frontend/src/pages/TimeEstimationPage.tsx

```diff
@@ -138,9 +153,30 @@ export const TimeEstimationPage = () => {
   };
 
   const onDurationChange = (duration: string) => {
-    setMilestoneDuration(duration);
+    const milestone = selectedMilestoneId;
+    const time = parseFloat(duration);
+    if (milestone !== undefined && roadmap) {
+      dispatch(
+        versionsActions.setTimeEstimate({
+          roadmapId: roadmap!.id,
+          id: milestone,
+          estimate: time > 0 ? time : undefined,
+        }),
+      );
+    }
   };
 
+  const milestoneDuration = useCallback(
+    () =>
+      roadmap && selectedMilestoneId !== undefined
+        ? timeEstimates.find(
+            ({ roadmapId, id }) =>
+              roadmapId === roadmap!.id && id === selectedMilestoneId,
+          )?.estimate
+        : undefined,
+    [roadmap, selectedMilestoneId, timeEstimates],
+  );
+
```

Comment:
useCallback ei taida olla tässä tarpeen (käytetään kun passataan callbackfunktiota jollekkin child-komponentille, eikä aina silloinkaan tarvita)

Reply:
Totta, poistin nyt useCallback():in